src/components/ProgressBar/ProgressBar.stories.tsx

```diff
@@ -0,0 +1,136 @@
+import { BADGE } from '@geometricpanda/storybook-addon-badges';
+import type { StoryObj, Meta } from '@storybook/react';
+import React from 'react';
+import { ProgressBar } from './ProgressBar';
+import Button from '../Button';
+import ButtonGroup from '../ButtonGroup';
+
+export default {
+  title: 'Molecules/Progress/ProgressBar',
+  component: ProgressBar,
+  args: {
+    label: 'Progress Bar Label',
+    max: 100,
+    segmentCount: 2,
+    segmentValue: 10,
+  },
+  parameters: {
+    badges: [BADGE.BETA],
+  },
+  decorators: [
+    (Story) => (
+      <div style={{ backgroundColor: 'white', padding: '0.5rem' }}>
+        <Story />
+      </div>
+    ),
+  ],
+} as Meta<Args>;
+
+type Args = React.ComponentProps<typeof ProgressBar>;
+
+export const Default: StoryObj<Args> = {};
+
+export const WithCustomCaption: StoryObj<Args> = {
+  args: {
+    caption: '20 %',
+  },
+};
+
+export const Empty: StoryObj<Args> = {
+  args: {
+    segmentCount: 0,
+  },
+};
+
+export const Complete: StoryObj<Args> = {
+  render: () => (
+    <div>
+      <ProgressBar
+        label="Progress Bar Label"
+        max={100}
+        segmentCount={10}
+        segmentValue={10}
+      />
+      <ProgressBar
+        label="Progress Bar Label"
+        max={100}
+        segmentCount={1}
+        segmentValue={100}
+      />
+    </div>
+  ),
+};
+
+const InteractiveProgressBar = () => {
```

Comment:
Would it be possible to use storybook controls to build the test interface? if we add `args` to render and spread that on this component, for example?

Reply:
Some of the args are accounted for via input controls 
![image](https://user-images.githubusercontent.com/86632227/206268539-9b7d9f4e-d528-4438-a6ba-b468978a3c4a.png)
and the purpose of this story is to show how the state of the progress bar can be managed in code, as the other stories provide controls via `args` 

However, I can pass the other props, i.e. `label`, `className`, and `caption` if that provides extra value